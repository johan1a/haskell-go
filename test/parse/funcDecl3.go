package main;

func f (x int ) int {
    return x
}

func main() {
    var k int = f(2)
    print(k)
}
