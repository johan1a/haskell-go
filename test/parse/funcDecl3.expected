SourceFile (Package "main") [TopLevelDecl2 (FunctionDecl2 "f" (Signature2 [ParameterDecl2 [IdDecl "x"] (TypeName (TypeNameIdentifier "int"))] (Result2 (TypeName (TypeNameIdentifier "int")))) (Block [ReturnStmt (IdUse "x")])),TopLevelDecl2 (FunctionDecl2 "main" (Signature1 []) (Block [DeclarationStmt (VarDecl [IdDecl "k"] (TypeName (TypeNameIdentifier "int")) [Call "f" [Num 2]]),SimpleStmt (ExpressionStmt (PrintCall [IdUse "k"]))]))]
